### PR TITLE
Fix compilation issue with scheduler extension

### LINKED ISSUE #5574: Scheduler extension fails compilation
I'm seeing the following exception when building Quarkus:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0-jboss-2:compile (default-compile) on project quarkus-scheduler: Compilation failure: Compilation failure: 
[ERROR] quarkus/quarkus/extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/DelegatingThreadPool.java:[9,18] package org.quartz does not exist
[ERROR] quarkus/quarkus/extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/DelegatingThreadPool.java:[10,24] package org.quartz.simpl does not exist
[ERROR] quarkus/quarkus/extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/DelegatingThreadPool.java:[11,22] package org.quartz.spi does not exist
[ERROR] quarkus/quarkus/extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/DelegatingThreadPool.java:[18,46] cannot find symbol
[ERROR]   symbol: class ThreadPool
[ERROR] quarkus/quarkus/extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/DelegatingThreadPool.java:[22,13] cannot find symbol
[ERROR]   symbol:   class ThreadPool
[ERROR]   location: class io.quarkus.scheduler.runtime.DelegatingThreadPool
[ERROR] quarkus/quarkus/extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/DelegatingThreadPool.java:[40,37] cannot find symbol
[ERROR]   symbol:   class SchedulerConfigException
[ERROR]   location: class io.quarkus.scheduler.runtime.DelegatingThreadPool
[ERROR] quarkus/quarkus/extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/DelegatingThreadPool.java:[74,68] cannot find symbol
[ERROR]   symbol:   class ThreadPool
[ERROR]   location: class io.quarkus.scheduler.runtime.DelegatingThreadPool
[ERROR] quarkus/quarkus/extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/DelegatingThreadPool.java:[109,41] cannot find symbol
[ERROR]   symbol:   class SchedulerConfigException
[ERROR]   location: class io.quarkus.scheduler.runtime.DelegatingThreadPool.EnhancedQueueExecutorBackedPool
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :quarkus-scheduler
```
I have tried both with the Maven version which I've locally `3.6.1` and the `mvnw` version `3.5.4` and it fails for both.
```
./mvnw -version
Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-18T00:03:14+05:30)
Maven home: /.m2/wrapper/dists/apache-maven-3.5.4-bin/4lcg54ki11c6mp435njk296gm5/apache-maven-3.5.4
Java version: 1.8.0_172, vendor: Oracle Corporation, runtime: /Library/Java/JavaVirtualMachines/jdk1.8.0_172.jdk/Contents/Home/jre
...
```

```
mvn -version
Apache Maven 3.6.1 (d66c9c0b3152b2e69ee9bac180bb8fcc8e6af555; 2019-04-05T00:30:29+05:30)
Maven home: /maven/apache-maven-3.6.1
Java version: 1.8.0_172, vendor: Oracle Corporation, runtime: /Library/Java/JavaVirtualMachines/jdk1.8.0_172.jdk/Contents/Home/jre
...
```

### PR DESCRIPTION
**Pull request by jaikiran**
Fixes https://github.com/quarkusio/quarkus/issues/5574

### CODE REVIEW

File: extensions/scheduler/runtime/pom.xml [hunk 1]
```diff
@@ -18,6 +18,10 @@
       <groupId>io.quarkus</groupId>
       <artifactId>quarkus-arc</artifactId>
     </dependency>
+    <dependency>
```
Comment (machi1990) [inline extensions/scheduler/runtime/pom.xml]: This does not seem right. 

We now have the `quartz` extension as part of https://github.com/quarkusio/quarkus/pull/5481 

Can you rebase with master and do a full build. I am not seeing any errors.

Comment (jaikiran) [reply extensions/scheduler/runtime/pom.xml]: This was against latest master (commit `5dca04d5e45d19a67655b67f4fc1dbf84ca68edd`). I'll redo again and see if there's something more going on.

Comment (jaikiran) [reply extensions/scheduler/runtime/pom.xml]: Hello @machi1990, sorry about the noise. It looks like I had a class called `DelegatingThreadPool` locally (not committed) which was lying around from my previous attempt at experimenting with using a custom thread pool for quartz. 

This is what is now failing compilation. Sorry about the noise. I'll close this one.

Comment (machi1990) [reply extensions/scheduler/runtime/pom.xml]: @jaikiran thanks for confirming :-)

**Pull request closed without merging**